src/App.js

```diff
@@ -1,17 +1,110 @@
 import React from 'react';
 import './App.css';
+import ClassNames from 'classnames';
+import { TodoList } from './componenets/TodoList';
 
-import users from './api/users';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-function App() {
-  return (
+class App extends React.Component {
+  state = {
+    todos: todosFromServer.map(todo => ({
+      ...todo,
+      user: usersFromServer.find(user => user.id === todo.userId),
+    })),
+    isHidden: true,
+    errorMessage: '',
+  }
+
+  timeToShowError = false;
```

Comment:
why not in state?

Reply:
because it didnt work in state. I was looking for bug 2 hours and then I understood that there are bug in react

Reply:
what kind of bug was it?

Reply:
i dont know, but calling of set state didnt work

Reply:
https://youtu.be/FiLo9wXdehw

Reply:
this is because async setState